openpype/modules/sync_server/launch_hooks/pre_copy_last_published_workfile.py

```diff
@@ -184,3 +185,22 @@ def execute(self):
         self.data["last_workfile_path"] = local_workfile_path
         # Keep source filepath for further path conformation
         self.data["source_filepath"] = last_published_workfile_path
+
+        # Get a make resources directory
+        resources_dir = os.path.join(
+            os.path.dirname(local_workfile_path), 'resources'
+        )
+        if not os.path.exists(resources_dir):
+            os.mkdir(resources_dir)
+
+        # Copy resources to the local resources directory
+        for file in workfile_representation['files']:
+            resource_path = re.sub(
+                r"\{root\[main\]\}", str(anatomy.roots['main']), file['path']
+            )
+            # Only copy if the resource file exists, and it's not the workfile
+            if (
+                os.path.exists(resource_path)
+                and resource_path != last_published_workfile_path
+            ):
+                shutil.copy(resource_path, resources_dir)
```

Comment:
Wouldn't this be quite dangerous in that it could override existing resources in the workdir if they did already exist? Not sure what the best UX is but this feels rather dangerous if that's the case. Should it either backup those files or maybe log a warning and skip the copy in that case?

Reply:
As this is hook is executed when no workfile is locally available, the only case I can think of in which that could happen is if a user goes into their task folder to delete all their workfiles but leaves the resource folder. Or if a user manually creates the task folder, the resource folder and manually copies the correct resources into it. But better safe than sorry. I think for now skipping with a warning might be more suitable, because local resource folders are not versioned so backing up resources would raise the question of "but what happens if the backup gets overwritten".

Reply:
I think it should check if checksum is the same. If not, it should add `.old` to files with same name already present in directory (and if a `my_name.old` already exists, do `.old01`...).